### PR TITLE
Disable IP spoofing on outbound

### PR DESCRIPTION
**Pull request by howardjohn**
This is not needed at all, as the ztunnel now always runs in the same
network as the workload (both inpod and dedicated).

The main part of this effort was the testing. The 'dedicated' mode is
not really testing 'dedicated mode', but rather than legacy geneve
stuff. This PR moves it closer to testing a real world setup (which runs
like an Istio sidecar/VM).

The challenge here is setting up iptables. For real world, we just run
the process and ztunnel as different namespaces. I couldn't find a way
to do that in our case, since we have one thread and some user namespace
magic (the latter was the bigger problem). To workaround this, I use a
mark, which is not compiled in by default.

The end result is we have two modes (in tests and real world):
* In pod mode
* Dedicated mode, with ztunnel running directly next to the app in the
  same network, with iptables rules doing traffic capture

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -492,6 +496,7 @@ pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
         inpod_port_reuse: parse_default(INPOD_PORT_REUSE, true)?,
         inpod_mark: parse_default(INPOD_MARK, DEFAULT_INPOD_MARK)?,
         fake_self_inbound: false,
+        testing_dedicated_mark: 0,
```
Comment (bleggett) [inline src/config.rs:499]: I am relatively sure that even non-testing usages of `dedicated` mode might need this mark, so TBH I am fine just making it `dedicated_mark` and setting it either via env or via `DEFAULT_DEDICATED_MARK` like we do with inpod above.

Alternatively - we just rename `inpod_mark` to `proxy_mark`, default to `1337` (current inpod default + also the current sidecar/envoy default) and use the same config val under both modes.


tl;dr I think this will be necessary for `dedicated` mode outside of just tests so we might as well formalize it? thoughts?

File: src/proxy/outbound.rs [hunk 2]
```diff
@@ -257,15 +249,8 @@ impl OutboundConnection {
         req: &Request,
         connection_stats: &ConnectionResult,
     ) -> Result<(), Error> {
-        // Create a TCP connection to upstream
-        // We do not need spoofing for inbound
-        let local = if self.enable_orig_src && self.pi.cfg.proxy_mode != ProxyMode::Shared {
```
Comment (bleggett) [inline src/proxy/outbound.rs:262]: :+1:

File: src/test_helpers/linux.rs [hunk 3]
```diff
@@ -142,16 +145,26 @@ impl WorkloadManager {
             local_ip: Some(ns.ip()),
             inpod_uds,
             proxy_mode,
+            testing_dedicated_mark: if proxy_mode == ProxyMode::Dedicated {
```
Comment (bleggett) [inline src/test_helpers/linux.rs:148]: Yeah - in ref to above comments our `inpod` and `dedicated_testing` marks are both 1337 - so maybe it should just be a single (overridable) "mark" config value, and we just use it differently according to ProxyMode

File: src/config.rs (hunk 1 above)
Comment (howardjohn) [reply src/config.rs:499]: I was on the fence, went with test only since I wasn't sure a concrete use case where running as a different UID/GID wasn't viable. Not opposed if there is (or if we think its simpler to just make it a real thing -- with the inpod_mark collapsing it may just be quite a bit simpler to do so)

Comment (howardjohn) [reply src/config.rs]: ok, I made the change

**Comment by howardjohn** [comment]
I added a cherrypick label since there is a small chance this fixes the obscure AWS issues we have seen.

**Review by bleggett — APPROVED** [review]

**Review by jacob-delgado — APPROVED** [review]
